internal/pkg/agent/application/upgrade/rollback.go

```diff
@@ -108,14 +109,18 @@ func Cleanup(log *logger.Logger, currentHash string, removeMarker bool, keepLogs
 
 // InvokeWatcher invokes an agent instance using watcher argument for watching behavior of
 // agent during upgrade period.
-func InvokeWatcher(log *logger.Logger) error {
+func InvokeWatcher(log *logger.Logger, targetHash string) error {
 	if !IsUpgradeable() {
 		log.Debug("agent is not upgradable, not starting watcher")
 		return nil
 	}
 
-	versionedHome := paths.VersionedHome(paths.Top())
-	cmd := invokeCmd(versionedHome)
+	if targetHash == "" {
+		// when targetHash is not provided then we use the current running hash
+		targetHash = release.ShortCommit()
+	}
+	hashedDir := fmt.Sprintf("%s-%s", agentName, targetHash)
+	cmd := invokeCmd(filepath.Join(paths.Top(), "data", hashedDir))
```

Comment:
This clashes with https://github.com/elastic/elastic-agent/issues/2579
Since by the time we invoke the watcher we already switched the symlink, can't we just call the watcher from home ?